webapi/src/main/scala/org/knora/webapi/responders/admin/ProjectsResponderADM.scala

```diff
@@ -438,18 +455,47 @@ final case class ProjectsResponderADMLive(
    *
    * @param id  the project's identifier (IRI / shortcode / shortname / UUID)
    *
-   * @return [[ProjectRestrictedViewSettingsGetResponseADM]]
+   * @return [[ProjectRestrictedViewSettingsResponseADM]]
    */
   override def projectRestrictedViewSettingsGetRequestADM(
     id: ProjectIdentifierADM
-  ): Task[ProjectRestrictedViewSettingsGetResponseADM] =
+  ): Task[ProjectRestrictedViewSettingsResponseADM] =
     projectRestrictedViewSettingsGetADM(id)
       .flatMap(ZIO.fromOption(_))
       .mapBoth(
         _ => NotFoundException(s"Project '${getId(id)}' not found."),
-        ProjectRestrictedViewSettingsGetResponseADM
+        ProjectRestrictedViewSettingsResponseADM
       )
 
+  /**
+   * Sets project's restricted view settings.
+   *
+   * @param iri the project's iri,
+   * @param user  requesting user,
+   * @param size  value to be set,
+   * @return [[ProjectRestrictedViewSettingsResponseADM]].
+   */
+  override def setProjectRestrictedViewSettings(
+    iri: Iri.ProjectIri,
+    user: UserADM,
+    size: RestrictedViewSize
+  ): Task[ProjectRestrictedViewSizeResponseADM] = {
+    val id = ProjectIdentifierADM.IriIdentifier(iri)
+    for {
+      _ <- getProjectFromCacheOrTriplestore(id)
+             .flatMap(ZIO.fromOption(_))
+             .orElseFail(NotFoundException(s"Project '${getId(id)}' not found."))
```

Comment:
Instead of `.flatMap(ZIO.fromOption(_)).orElseFail(` ZIO provides `.someOrElseFail(`

Reply:
guess I did.